### PR TITLE
Enable i64 bit indexing for binary MLIR kernels

### PR DESCRIPTION
**Pull request by kushanam**
Enabling experimental i64 bit indexing for binary milr kernels
@frgossen

### CODE REVIEW

File: tensorflow/core/kernels/mlir_generated/BUILD [hunk 1]
```diff
@@ -220,42 +220,82 @@ tf_kernel_library(
...
-        ":gpu_pow_kernels",
-        ":gpu_right_shift_kernels",
-        ":gpu_select_v2_kernels",
-        ":gpu_squared_difference_kernels",
-        ":gpu_sub_kernels",
-        ":gpu_xdivy_kernels",
-        ":gpu_xlog1py_kernels",
-        ":gpu_xlogy_kernels",
-        ":gpu_zeta_kernels",
+
```
Comment (frgossen) [inline tensorflow/core/kernels/mlir_generated/BUILD:225]: nit: whitespace

File: tensorflow/core/kernels/mlir_generated/BUILD [hunk 2]
```diff
@@ -220,42 +220,82 @@ tf_kernel_library(
     tags = ["manual"],
     deps = if_mlir_generated_gpu_kernels_enabled([
         ":base_gpu_op",
-        ":gpu_add_v2_kernels",
-        ":gpu_atan2_kernels",
-        ":gpu_bitwise_and_kernels",
-        ":gpu_bitwise_or_kernels",
-        ":gpu_bitwise_xor_kernels",
+
```
Comment (frgossen) [inline tensorflow/core/kernels/mlir_generated/BUILD:223]: nit: whitespace

**Review by frgossen — CHANGES_REQUESTED** [review]
Nice! Can you clarify in the PR name / description that this adds i64-indexing support for *binary* kernels behind an experimental flag?

File: tensorflow/core/kernels/mlir_generated/BUILD (hunk 2 above)
Comment (kushanam) [reply tensorflow/core/kernels/mlir_generated/BUILD:223]: > Nice! Can you clarify in the PR name / description that this adds i64-indexing support for _binary_ kernels behind an experimental flag?

done.

File: tensorflow/core/kernels/mlir_generated/BUILD (hunk 1 above)
Comment (kushanam) [reply tensorflow/core/kernels/mlir_generated/BUILD]: done

**Review by frgossen — COMMENTED** [review]
Will add this kernels one by one to avoid merge conflict.

**Comment by gbaned** [comment]
Hi @kushanam Can you please check @frgossen's comments and keep us posted ? Thank you!

**Comment by kushanam** [comment]
> Hi @kushanam Can you please check @frgossen's comments and keep us posted ? Thank you!

Thank @gbaned. As pointed out by Frederik, the kernels will be added one by one. I want to leave this open for reference if you don't mind.

**Comment by kushanam** [comment]
reopening the PR for further integrating kernels.

**Comment by kushanam** [comment]
This will be split into chunk PRs for maintainability.

**Pull request closed without merging**